clusterloader2/pkg/measurement/common/probes/probes.go

```diff
@@ -0,0 +1,338 @@
+		return
+	}
+	klog.Info("Stopping probes...")
+	k8sClient := p.framework.GetClientSets().GetClient()
+	if err := client.DeleteNamespace(k8sClient, probesNamespace); err != nil {
+		klog.Errorf("error while deleting %s namespace: %v", probesNamespace, err)
+	}
+	if err := client.WaitForDeleteNamespace(k8sClient, probesNamespace); err != nil {
+		klog.Errorf("error while waiting for %s namespace to be deleted: %v", probesNamespace, err)
+	}
+	*p = *createProbesMeasurement().(*probesMeasurement) // Reset measurement state.
+}
+
+// String returns string representation of this measurement.
+func (p *probesMeasurement) String() string {
+	return name
+}
+
+func (p *probesMeasurement) initialize(config *measurement.MeasurementConfig) error {
+	replicasPerProbe, err := util.GetInt(config.Params, "replicasPerProbe")
+	if err != nil {
+		return err
+	}
+	p.framework = config.ClusterFramework
+	p.replicasPerProbe = replicasPerProbe
+	p.templateMapping = map[string]interface{}{"Replicas": replicasPerProbe}
+	return nil
+}
+
+func (p *probesMeasurement) start(config *measurement.MeasurementConfig) error {
+	if !p.startTime.IsZero() {
+		return fmt.Errorf("measurement %s cannot be started twice", name)
+	}
+	if err := p.initialize(config); err != nil {
+		return err
+	}
+	k8sClient := p.framework.GetClientSets().GetClient()
+	if err := client.CreateNamespace(k8sClient, probesNamespace); err != nil {
+		return err
+	}
+	if err := p.createProbesObjects(); err != nil {
+		return err
+	}
+	if err := p.waitTillProbesReady(); err != nil {
+		return err
+	}
+	p.startTime = time.Now()
+	return nil
+}
+
+func (p *probesMeasurement) gather(params map[string]interface{}) (*probesSummary, error) {
+	if p.startTime.IsZero() {
+		return nil, fmt.Errorf("measurement %s has not been started", name)
+	}
+	thresholds, err := parseThresholds(params)
+	if err != nil {
+		return nil, err
+	}
+	measurementEnd := time.Now()
+	var probeSummaries []*probeSummary
+	for probeName, queryTmpl := range p.probeNameToPrometheusQueryTmpl {
+		query := prepareQuery(queryTmpl, p.startTime, measurementEnd)
+		samples, err := executePrometheusQuery(p.framework.GetClientSets().GetClient(), query, measurementEnd)
+		if err != nil {
+			return nil, err
+		}
+		probe := &probeSummary{name: probeName}
+		for _, sample := range samples {
+			quantile, err := strconv.ParseFloat(string(sample.Metric["quantile"]), 64)
+			if err != nil {
+				return nil, err
+			}
+			latency := time.Duration(float64(sample.Value) * float64(time.Second))
+			probe.latency.SetQuantile(quantile, latency)
+		}
+		if threshold, ok := thresholds[probeName]; ok {
+			if err := probe.latency.VerifyThreshold(threshold); err != nil {
+				err = errors.NewMetricViolationError(probe.name, err.Error())
+				klog.Errorf("%s: %v", p, err)
+			}
+		}
+		probeSummaries = append(probeSummaries, probe)
+	}
+	return &probesSummary{probeSummaries: probeSummaries}, err
+}
+
+func (p *probesMeasurement) createProbesObjects() error {
```

Comment:
Seems really similar to this one:
https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/pkg/prometheus/prometheus.go#L134

Can we refactor and reuse?

Reply:
Done in https://github.com/kubernetes/perf-tests/pull/482